tests/test_select.py

```diff
@@ -99,6 +100,23 @@ def test_select_default_throw_on_missing(
         OmegaConf.select(cfg, key, default=default, throw_on_missing=True)
 
 
+@pytest.mark.parametrize(
+    "cfg, key",
+    [
+        pytest.param(["???"], "0", id="missing"),
+    ],
+)
+def test_select_default_throw_on_missing_list(
+    cfg: Any,
+    key: Any,
+) -> None:
+    cfg = OmegaConf.create(cfg)
+
+    # throw on missing still throws if default is provided
+    with pytest.raises(MissingMandatoryValue):
+        OmegaConf.select(cfg, key, default=0, throw_on_missing=True)
+
+
```

Comment:
looks like this is covered by the first test you changed?

Reply:
I think that this test covers the `throw_on_missing=True` case, the other one does not.

Reply:
No, the `list:missing` test I added is a success case when `throw_on_missing` is False (it was working before as well, but it wasn't tested, I just added it for better test coverage)

Reply:
How is it a success case if the test body has this?
```
with pytest.raises(MissingMandatoryValue):
   ...
```

Reply:
Trying to clear up the confusion:
* I thought by "first test you changed" you meant `test_select()`: I indeed added a new test here (id `list:missing`) to test the case where we select a missing value with `throw_on_missing` set to False (which worked before, and still works here)
* The test with `with pytest.raises(MissingMandatoryValue):` is `test_select_default_throw_on_missing`: this tests the case where `throw_on_missing` is True. The new test I added here would have failed before this PR.